modules/packages/LinearAlgebra.chpl

```diff
@@ -221,6 +222,265 @@ class LinearAlgebraError : Error {
     }
 }
 
+pragma "no doc"
+/*
+This class promotes lazy computation of required
+Helper matrices to evaluate matrix exponential.
+
+High precision of expm requires smaller computations,
+hence we do a lazy computation depending on the norm value
+of the matrix.
+*/
+class ExpmPadeHelper {
+  type matType;
+  type eltType;
+  var A : matType;
+  var A2 : matType;
+  var A4 : matType;
+  var A6 : matType;
+  var A8 : matType;
+  var A10 : matType;
+  var Exactd4 : eltType;
```

Comment:
A mathematical nit - norms are always real, even if the matrices are complex, so `eltType` is always `real`.

Reply:
Ooh, I'm sorry I missed that.